src/shared/install.c

```diff
@@ -1755,6 +1757,10 @@ int unit_file_mask(
                 _cleanup_free_ char *path = NULL;
                 int q;
 
+                r = install_info_discover(scope, &c, &paths, *i, SEARCH_LOAD, &j);
+                if (r < 0)
+                        return r;
```

Comment:
This doesn't look right. It really should only print a warning, but not fail completely, but the code above makes the whole thing fail fully and fatally...